api/v1alpha/src/lib/metaManager.js

```diff
@@ -140,11 +155,16 @@ async function performDatasetMetadataUpdate(projectId, datasetId, accounts) {
  * @param  {} accounts
  */
 async function performTableMetadataUpdate(projectId, datasetId, tableId, accounts) {
-    const accessTypes = ["user", "group"];
-    const viewerRole = 'roles/bigquery.dataViewer';
     console.log(`Begin metadata update for table: ${datasetId}.${tableId}`);
-    let isDirty = false;
     const bigqueryUtil = new BigQueryUtil(projectId);
+    const exists = await bigqueryUtil.tableExists(datasetId, tableId);
+    if (!exists) {
+        console.warn(`Skipping metadata update for non-existant table: ${datasetId}.${tableId}`);
+        return false;
+    }
+    const accessTypes = ["user", "group", "serviceAccount"];
+    const viewerRole = 'roles/bigquery.dataViewer';
```

Comment:
Can an end-user have another viewerRole or elevated permissions? i.e. A BQ admin with **roles/bigquery.admin** which would include all 'bigquery.*' permissions.

Reply:
Currently, in datashare we only have the concept of granting reader access. If we introduced other role permissions it would require changes across the board.